### ISSUE TITLE
Bug: `select-fn->fn` does not realize the result if using `identity` as xform

### ISSUE THREAD

**Issue by qnkhuat** [camsaul/toucan2#144]
```
(methodical/defmethod t2/table-name :card [_model] :report_card)

(t2/select-fn->fn :id identity :card)
=>
{8 ^toucan2.jdbc.row.TransientRow {:id 8},
 9 ^toucan2.jdbc.row.TransientRow {:id 9},
 10 ^toucan2.jdbc.row.TransientRow {:id 10},
 6 ^toucan2.jdbc.row.TransientRow {:id 6},
 11 ^toucan2.jdbc.row.TransientRow {:id 11},
 12 ^toucan2.jdbc.row.TransientRow {:id 12}}
```

if using identity as xform, the result is a transientrow, not an instance.

Current work around

`(t2/select-fn->fn :id (comp identity t2.realize/realize :card)`

I'm not sure whether this is a bug or not, but would like to know if we could avoid this footgun.
Ơi

**Comment by camsaul** [comment]
I think maybe we can have `select-fn->fn` and the like make sure its results are realized **after** applying the user-supplied transforms, e.g. instead of `(map identity)` we do `(comp (map identity) (map realize))`

**Comment by camsaul** [comment]
Fixed in `1.0.530`
